rasa/core/featurizers/tracker_featurizers.py

```diff
@@ -450,9 +451,15 @@ def prediction_states(
         ]
         # TODO there is no prediction support for e2e input right now, therefore
         #  temporary remove TEXT features from USER state during prediction
+        #  also temporary remove full retrieval intent
         for states in trackers_as_states:
             for state in states:
                 if state.get(USER, {}).get(TEXT):
                     del state[USER][TEXT]
 
+                # remove full intent, only keep base intent
```

Comment:
why do we need to remove the full intent?

Reply:
@kalkbrennerei were we supposed to detect if the `intent` is a `retrieval intent` here, and take only the `base intent` from it? If yes, then:
- we should adjust the comment, since `temporary remove full retrieval intent` and `remove full intent` sound scary 🙀 
- `INTENT_MESSAGE_PREFIX` -> `RESPONSE_IDENTIFIER_DELIMITER` **or** check my suggestion below, we probably don't even need to check anything manually 👍

Reply:
@dakshvar22 What do you think about this change to the `rasa/core/featurizers/tracker_featurizers.py`?
Do we also need to do that for both `FullDialogueTrackerFeaturizer` and `MaxHistoryTrackerFeaturizer`?

Reply:
Just so that I understand this correctly, a retrieval intent with its response key can only be present in the tracker during these E2E test stories because they are written in a way to have them there? I don't think they can be present during inferencing/training because the response key is never added to the intent present in the tracker.
From that perspective, yes making this change makes sense in both `FullDialogueTrackerFeaturizer` and `MaxHistoryTrackerFeaturizer`. @degiz Also, I agree your suggestion below is a better way to do it since it reuses the method for separating the intent from response key.
As an alternate approach, would it be possible to create the intents inside the trackers without the response key for prediction in core? They are still needed to validate if NLU predicted both intent and response key correctly. Then we don't need to have this logic here.

Reply:
maybe I can add a bit more context on that one:

- for the **why**: I added this because the prediction was confused by seeing the `full retrieval intent` and this would lead to wrong `action predictions`. This happened to me when using `rasa-demo` with a test story that contains `retrieval intents`. Shall I provide more info on that to reproduce?
- for the **where**: I assumed that this was related to the missing prediction support for `e2e input`, that's why i moved it there. Please feel free to suggest to move it to another position in the code!!
- also note that this is about the `full retrieval intent` of the `gold intent` and also only removed temporarily for the `prediction`

I'm happy to hear about your opinions on this @tmbo @degiz